### PR TITLE
Add ability to use a yaml and csv for configuration of the export

### PR DESCRIPTION
**Pull request by mfussi66**
This PR adds the possibility to configure how the export to urdf is ran with a yaml file and a csv file.

This is the first stepping stone for replicating the same features of https://github.com/robotology/simmechanics-to-urdf.

The yaml tags supported are:
- root
- robotName
- rename
- linkFrames
- reverseRotationAxis
- color
- exportedFrames
- assignedMasses
- assignedInertias
- XMLBlobs
- sensors
- forceTorqueSensors
- originXYZ
- originRPY
- scale
-  filenameformatchangeext
- forcelowercase
- stringToRemoveFromMeshFileName

The csv columns supported are:
- upper limit
- lower limit

Result:
262405284-52502224-de59-45cd-9de3-433847a63c91

At the moment only ergocub is supported.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ Generate URDF model from CREO mechanisms.
 Right now the `creo2urdf` plugin needs its dependencies to be compiled and linked **statically**:
 - Download [vcpkg](https://github.com/microsoft/vcpkg): `git clone https://github.com/microsoft/vcpkg`
 - Bootstrap vcpkg: `.\vcpkg\bootstrap-vcpkg.bat`
-- Run `[path to vcpkg]/vcpkg install --triplet x64-windows-static-md eigen3 libxml2 assimp idyntree[core,assimp]`
+- Run `[path to vcpkg]/vcpkg install --triplet x64-windows-static-md eigen3 yaml-cpp libxml2 assimp idyntree[core,assimp]`
```
Comment (traversaro) [inline README.md:27]: `rapidcsv` is missing here?

File: src/creo2urdf/src/Sensorizer.cpp [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+    }
+}
+
+std::vector<std::string> Sensorizer::buildFTXMLBlobs()
+{
+    std::vector<std::string> ft_xml_blobs;
+
+    for (const auto& ft : ft_sensors)
+    {
+        //std::string filename = ft.jointName + ".xml";
```
Comment (traversaro) [inline src/creo2urdf/src/Sensorizer.cpp:106]: Leftover?

File: src/creo2urdf/src/Sensorizer.cpp [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+                xmlAddChild(node, node_xmlblob);
+        }
+
+        xmlOutputBufferPtr gazebo_doc_buffer = xmlAllocOutputBuffer(NULL);
+        xmlNodeDumpOutput(gazebo_doc_buffer, doc, root_node, 0, 1, NULL);
+        ft_xml_blobs.push_back(string((char*)xmlBufContent(gazebo_doc_buffer->buffer)));
+
+        xmlOutputBufferClose(gazebo_doc_buffer);
+
+        //xmlSaveFormatFile(filename.c_str(), doc, 1);
```
Comment (traversaro) [inline src/creo2urdf/src/Sensorizer.cpp:164]:
```suggestion
        //xmlSaveFormatFile(filename.c_str(), doc, 1);
```

Leftover?

File: src/creo2urdf/src/Sensorizer.cpp [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+        }
+
+        xmlOutputBufferPtr doc_buffer = xmlAllocOutputBuffer(NULL);
+        xmlNodeDumpOutput(doc_buffer, doc, root_node, 0, 1, NULL);
+        xml_blobs.push_back(string((char*)xmlBufContent(doc_buffer->buffer)));
+
+        xmlOutputBufferClose(doc_buffer);
+
+        xmlFreeDoc(doc);
+        xmlCleanupParser();
```
Comment (traversaro) [inline src/creo2urdf/src/Sensorizer.cpp:275]: Please remove, this function needs to be called only before exit (see https://gnome.pages.gitlab.gnome.org/libxml2/devhelp/libxml2-parser.html#xmlCleanupParser and https://gnome.pages.gitlab.gnome.org/libxml2/devhelp/libxml2-parser.html#xmlCleanupParser).

File: src/creo2urdf/src/Sensorizer.cpp [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+        xmlNewProp(node, BAD_CAST "xyz", BAD_CAST trf.getPosition().toString().c_str());
+
+        doc_buffer = xmlAllocOutputBuffer(NULL);
+        xmlNodeDumpOutput(doc_buffer, doc, root_node, 0, 1, NULL);
+        xml_blobs.push_back(string((char*)xmlBufContent(doc_buffer->buffer)));
+
+        xmlOutputBufferClose(doc_buffer);
+
+        xmlFreeDoc(doc);
+        xmlCleanupParser();
```
Comment (traversaro) [inline src/creo2urdf/src/Sensorizer.cpp:296]: Please remove, this function needs to be called only before exit (see https://gnome.pages.gitlab.gnome.org/libxml2/devhelp/libxml2-parser.html#xmlCleanupParser and https://gnome.pages.gitlab.gnome.org/libxml2/devhelp/libxml2-parser.html#xmlCleanupParser).

File: src/creo2urdf/src/Validator.cpp [hunk 6]
```diff
@@ -50,7 +51,9 @@ bool Validator::assignCreoTransformToLink() {
         auto modelhdl = creo_session_ptr->RetrieveModel(pfcComponentFeat::cast(feat)->GetModelDescr());
         auto link_child_name = string(modelhdl->GetFullName());
 
-        std::tie(ret, H_child) = getTransformFromRootToChild(comp_path, modelhdl);
+        //std::tie(ret, H_child) = getTransformFromRootToChild(comp_path, modelhdl);
```
Comment (traversaro) [inline src/creo2urdf/src/Validator.cpp:54]: Leftover? Can we remove?

**Review by traversaro — CHANGES_REQUESTED** [review]
Minor changes, most can be ignored, except for the need to avoid calls to `xmlCleanupParser()` that are quite dangerous and can introduce problematic bugs.

File: src/creo2urdf/src/Sensorizer.cpp [hunk 7]
```diff
@@ -0,0 +1,300 @@
...
+        xmlOutputBufferPtr gazebo_doc_buffer = xmlAllocOutputBuffer(NULL);
+        xmlNodeDumpOutput(gazebo_doc_buffer, doc, root_node, 0, 1, NULL);
+        ft_xml_blobs.push_back(string((char*)xmlBufContent(gazebo_doc_buffer->buffer)));
+
+        xmlOutputBufferClose(gazebo_doc_buffer);
+
+        //xmlSaveFormatFile(filename.c_str(), doc, 1);
+
+        xmlFreeDoc(doc);
+        xmlCleanupParser();
```
[the remark this answers is not in the document]
Comment (mfussi66) [reply src/creo2urdf/src/Sensorizer.cpp]: Remove all occurrences of xmlCleanupParser.

File: CMakeLists.txt [hunk 8]
```diff
@@ -14,6 +14,11 @@ include(FeatureSummary)
 # Needed for https://github.com/robotology/idyntree/issues/1065
 find_package(Eigen3 REQUIRED)
 find_package(iDynTree REQUIRED)
+find_package(yaml-cpp REQUIRED)
+find_package(LibXml2 REQUIRED)
+
+find_path(RAPIDCSV_INCLUDE_DIRS "rapidcsv.h")
```
Comment (Nicogene) [inline CMakeLists.txt:20]: `find_package(rapidcsv)` isn't working?

Comment (mfussi66) [reply CMakeLists.txt:20]: Since it's header only, it seems to be the suggested way to include it.

Comment (traversaro) [reply CMakeLists.txt:20]: Apparently rapidcsv does not ship any `rapidcsv-config.cmake` file or similar.

Comment (mfussi66) [reply CMakeLists.txt:20]: I chose it for its extreme simplicity, but we can change it if preferred. However, I think it is possible to get the csv data directly from Creo (eg the joint limits), so at some point I think we could drop the csv support.

**Review by Nicogene — APPROVED** [review]

**Pull request merged**